src/elements/content-explorer/ContentExplorer.js

```diff
@@ -754,8 +743,9 @@ class ContentExplorer extends Component<Props, State> {
                 return item;
             });
         }
-
-        return { newCollection, newSelected };
+        this.setState({ currentCollection: newCollection, selected: newSelected }, () => {
```

Comment:
Can probably leave out the extra anonymous function and just pass in the callback reference directly